crates/polars-sql/src/sql_expr.rs

```diff
@@ -345,17 +353,68 @@ impl SQLExprVisitor<'_> {
         }
     }
 
+    /// Handle implicit temporal string comparisons.
+    ///
+    /// eg: "dt >= '2024-04-30'", or "dtm::date = '2077-10-10'"
+    fn convert_temporal_strings(&mut self, left: &Expr, right: &Expr) -> Expr {
+        if let (Some(name), Some(s), expr_dtype) = match (&left, &right) {
```

Comment:
`&left -> left` as we already borrow. Though I'd expect clippy to catch this one. Do we run it on this crate?

Reply:
Yup; it caught a few other minor things in this PR, heh